src/main/java/org/opensrp/web/controller/MultimediaController.java

```diff
@@ -142,7 +143,14 @@ public void downloadFiles(HttpServletResponse response, @PathVariable("entity-id
 			zipOutputStream.close();
 		} else {
 			// default to single profile image retrieval logic
-			downloadFileWithAuth(entityId, response);
+			Multimedia multimedia = multimediaService.findByCaseId(String.valueOf(entityId));
+			String extension = "";
+			int startingIndex;
+			if (multimedia != null && multimedia.getOriginalFileName() != null) {
+				startingIndex = multimedia.getOriginalFileName().indexOf('.');
+				extension = multimedia.getOriginalFileName().substring(startingIndex);
+			}
+			downloadFileWithAuth(entityId, response, extension);
```

Comment:
move this to a method add use it in `downloadFileByClientId` instead of passing null

Reply:
There seems a problem with this approach, since while saving if the image file name has extension `.jpeg `but its content type is of `image/jpeg` it modifies extension to ".jpg" and then persists file. Therefore, I am creating a new method that checks for file extension and then retrieves a file based on that. @githengi Does it sounds okay?

Reply:
yes sounds okay